### DISCUSSION TITLE
VSCode Extension: Tailwind Twin IntelliSense is no longer supported

### DISCUSSION THREAD

**Discussion by andymerskin** [ben-rogerson/twin.macro#798] [Ideas]
Hey there!

<img width="1348" alt="image" src="https://user-images.githubusercontent.com/758090/236955287-5cf564c0-7e4d-4710-882b-4f869a2d094a.png">

It looks like the popular [**Tailwind Twin IntelliSense**](https://github.com/lightyen/vscode-tailwindcss-twin)'s repo has been archived and is no longer being supported. Are there any plans to continue support officially, or write a new extension? Are there any other extensions that support Twin.Macro with the newer features from Tailwind CSS?

I recently discovered this wasn't be supported anymore since the extension was erroring out when I was using **differentiated nesting groups**, a newer feature: https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups

Thanks for all the hard work!

**Comment by ben-rogerson** [comment]
Hey Andy

Unfortunately I'm hesitant to take over the plugin because it would require a large amount of extra work to keep the extension up-to-date with syntax changes from tailwindcss.
 
Another idea may be to rewrite the plugin to import and use code from the official intellisense plugin in an effort to avoid the need for copy-paste updates. But obviously this can cause issues if the imported code changes too much from version to version.

The official intellisense extension can be used instead after a couple of settings updates to `tailwindCSS.experimental.classRegex` - https://github.com/ben-rogerson/twin.macro/discussions/227#discussion-38611
It's not perfect though as there's no support for custom twin features like variant groups:

```js
tw`sm:(pt-4 px-4)`
```

If you or anyone else would like to help with any updates or ideas please let me know - it would be great to keep the helpful extension up-to-date with the latest tailwind+twin features.
